### PR TITLE
make daemon_conn and msg_queue tal objects

### PR DESCRIPTION
**Pull request by rustyrussell**
This is more tal-idiomatic IMHO.  Though it means an extra allocation at startup time, I think it adds clarity.

### CODE REVIEW

File: common/daemon_conn.c [hunk 1]
```diff
@@ -73,23 +101,30 @@ static struct io_plan *daemon_conn_start(struct io_conn *conn,
...
-	dc->msg_in = NULL;
-	msg_queue_init(&dc->out, dc->ctx);
-	dc->msg_queue_cleared_cb = NULL;
-	dc->conn = io_new_conn(ctx, fd, daemon_conn_start, dc);
+	/* Harmless free loop if conn is being destroyed because dc freed */
+	tal_free(dc);
 }
 
-void daemon_conn_clear(struct daemon_conn *dc)
+struct daemon_conn *new_daemon_conn_(const tal_t *ctx, int fd,
```
Comment (cdecker) [inline common/daemon_conn.c]: Keeping with the `ccan` naming convention this should be `daemon_conn_new_`.

File: common/msg_queue.c [hunk 2]
```diff
@@ -1,18 +1,22 @@
...
 #include <common/msg_queue.h>
 #include <common/utils.h>
 #include <wire/wire.h>
 
-void msg_queue_init(struct msg_queue *q, const tal_t *ctx)
+struct msg_queue {
+	const u8 **q;
+};
+
+struct msg_queue *new_msg_queue(const tal_t *ctx)
```
Comment (cdecker) [inline common/msg_queue.c]: Nit; naming (`msg_queue_new`)

File: common/daemon_conn.c (hunk 1 above)
Comment (rustyrussell) [reply common/daemon_conn.c]: Yes.  Good point!

File: common/msg_queue.c (hunk 2 above)
Comment (rustyrussell) [reply common/msg_queue.c]: Ack...

**Comment by cdecker** [comment]
Rebased on top of `master` and squashed fixup. Will merge once Travis passes (and my plane lands) :-)

**Comment by cdecker** [comment]
No idea why `test_onchain_multihtlc_their_unilateral` seems to stall and not complete with `VALGRIND=0`, `DEVELOPER=1` on `ARCH=32`

**Comment by rustyrussell** [comment]
Looks like a rebase error ATM: the first commit does not belogn?

**Pull request merged**